test/generator/filler/test_retcon.py

```diff
@@ -50,3 +53,10 @@ def test_calculate_weights_for_output(capsys, blank_pickup):
         "  hints: []\n"
         "nodes: []\n\n"
     )
+
+
+def test_invalid_starting_items():
+    pickup_entry = MagicMock()
+    pickup_entry.start_case = StartingItemBehavior.CAN_NEVER_BE_STARTING
+
+    assert not should_be_starting_pickup(MagicMock(), MagicMock(), pickup_entry)
```

Comment:
this only tests the very first case of this function - can you add cases that achieve coverage on the other parts of this func?